text/1028-cli-hooks.rst

```diff
@@ -0,0 +1,159 @@
+::
+
+    Status: Draft
+    Type: Feature
+    Created: 2025-01-14
+    Authors: Victor Petrovykh <victor@edgedb.com>
+
+===================
+RFC 1028: CLI hooks
+===================
+
+This RFC discusses addition of hooks to CLI commands. In order to make it
+easier to setup and maintain some of the things dependent on the schema state
+we can introduce hooks for some of the schema altering commands.
+
+
+Motivation
+==========
+
+There are a number of tools that generate code based on the current schema
+(such as query-building tools, or query reflection tools). They need to be
+re-run after every schema change in order to keep the generated code in sync
+with the database. In larger projects the number of things that may need
+updating after a schema change is likely to grow. Migration hooks could make
+it easier to automate these tasks.
+
+There is also a longstanding need to implement some kind of mechanism for
+creating data fixtures. Unlike the migration hooks, fixtures are more likely
+to be run at project initialization or after a branch wipe command.
+
+
+Hooks Configuration
+===================
+
+All hooks are intended to be used within an active project, not for arbitrary
+CLI commands. Thus if a CLI command receives an explicit ``-I <instance>``
+option, the command should not trigger any project hooks even if the instance
+is linked to a project. Only commands that omit the instance and use the
+current project to determine the target instance are going to trigger hooks.
+
+The hooks will be described in the ``gel.toml`` file in the
```

Comment:
Can the table be named just `hooks` instead of `project-hooks`?

Currently, this project manifest contains only two options: `project.schema-dir` and `instance.server-version`.

Reply:
Given that the toml file is for projects, we can just name the table `hooks`, I guess. Everything here is for the project.